backend/database/versions/f6c4f66e8db6_update_for_collections.py

```diff
@@ -0,0 +1,164 @@
+"""update_for_collections
+
+Revision ID: f6c4f66e8db6
+Revises: 9023d0bf61ab
+Create Date: 2020-10-27 16:14:31.084060
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = "f6c4f66e8db6"
+down_revision = "9023d0bf61ab"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # For more information about the changes executed here, please visit the design document and the approved
+    # modifications to the database UML:
+    # https://github.com/chanzuckerberg/single-cell/blob/main/rfcs/0001-data-portal-architecture/text.md
+
+    # Create Visibility Enum
+    visibility_enum = postgresql.ENUM("PUBLIC", "PRIVATE", name="collectionvisibility")
+    visibility_enum.create(op.get_bind())
+
+    # Add visibility column
+    op.add_column(
+        "project", sa.Column("visibility", sa.Enum("PUBLIC", "PRIVATE", name="collectionvisibility"), default="PRIVATE")
+    )
+    op.add_column(
+        "dataset",
+        sa.Column("project_visibility", sa.Enum("PUBLIC", "PRIVATE", name="collectionvisibility"), default="PRIVATE"),
+    )
+    op.add_column(
+        "project_link",
+        sa.Column("project_visibility", sa.Enum("PUBLIC", "PRIVATE", name="collectionvisibility"), default="PRIVATE"),
+    )
+
+    # Map Status to Visibility Column
+    op.execute("UPDATE project SET visibility = 'PUBLIC' WHERE status = 'LIVE'")
+    op.execute("UPDATE project SET visibility = 'PRIVATE' WHERE status = 'EDIT'")
+    op.execute("UPDATE dataset SET project_visibility = 'PUBLIC' WHERE project_status = 'LIVE'")
+    op.execute("UPDATE dataset SET project_visibility = 'PRIVATE' WHERE project_status = 'EDIT'")
+    op.execute("UPDATE project_link SET project_visibility = 'PUBLIC' WHERE project_status = 'LIVE'")
+    op.execute("UPDATE project_link SET project_visibility = 'PRIVATE' WHERE project_status = 'EDIT'")
+
+    # Remove (id, status) Primary and Foreign Key Constraint
+    op.execute("ALTER TABLE project DROP CONSTRAINT project_pkey CASCADE")
```

Comment:
Since this is a constraint in this table, as well as a foreign constraint in other tables, using alter with cascade cleanly removed all of the constraints. The alternative is 3 drop_contraint commands with an order dependency.